sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -798,8 +798,13 @@ class SQLContext(@transient val sparkContext: SparkContext)
    * @group ddl_ops
    * @since 1.3.0
    */
-  def table(tableName: String): DataFrame =
-    DataFrame(this, catalog.lookupRelation(Seq(tableName)))
+  def table(tableName: String): DataFrame = {
+    table(new SqlParser().parseTableIdentifier(tableName))
+  }
+
+  private[sql] def table(tableIdent: TableIdentifier): DataFrame = {
```

Comment:
is this called anywhere? It would be great to remove this if possible, since this API is also accessible in Java, and Java users won't see this as a private API.